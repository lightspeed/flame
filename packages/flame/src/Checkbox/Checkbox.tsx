import * as React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { themeGet } from '@styled-system/theme-get';
import { layout, LayoutProps } from 'styled-system';
import { IconCheckmark } from '../Icon/Checkmark';
import { RadioLabel } from '../Radio';

const CheckboxLabel = RadioLabel;

const Wrapper = styled('div')`
  position: relative;
  line-height: 1rem;
`;

const CheckboxCheckmarkWrapper = styled('div')<LayoutProps & { indeterminate: boolean }>`
  position: relative;
  ${layout};
  flex: 0 0 auto;
  border-radius: ${themeGet('radii.radius-1')};
  background-color: transparent;
  border: solid 1px ${themeGet('checkboxStyles.borderColor')};
  box-sizing: border-box;
  overflow: hidden;
  cursor: default;
  background: ${themeGet('checkboxStyles.background')};
  margin-right: ${themeGet('space.2')};

  ${props =>
    props.indeterminate &&
    css`
      border-color: ${themeGet('checkboxStyles.indeterminate.border')(props)};
    `};
`;

CheckboxCheckmarkWrapper.defaultProps = {
  width: ['18px', '16px'],
  height: ['18px', '16px'],
};

const CheckboxIndeterminate = styled('div')`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 0.5rem;
  height: 0.125rem;
  border-radius: ${themeGet('radii.radius-1')};
  background: ${themeGet('checkboxStyles.indeterminate.background')};
`;

const StyledIcon = styled(IconCheckmark)<{ size: string }>`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  fill: ${themeGet('checkboxStyles.checked.color')};
`;

const CheckboxInput = styled('input')<{ indeterminate: boolean }>`
  position: absolute;
  ${layout}
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;

  &:checked + ${CheckboxCheckmarkWrapper} {
    background: ${themeGet('checkboxStyles.checked.background')};
    border-color: ${themeGet('checkboxStyles.checked.border')};

    ${CheckboxIndeterminate} {
      display: none;
    }
  }

  &:checked + div ${StyledIcon} {
    display: block;
  }

  &:not(:checked) + div ${StyledIcon} {
    display: none;
  }

  &:focus + ${CheckboxCheckmarkWrapper} {
    box-shadow: ${themeGet('checkboxStyles.focus.boxShadow')};
  }

  &[disabled] + ${CheckboxCheckmarkWrapper} {
    opacity: 0.5;
  }
`;

CheckboxInput.defaultProps = {
  // Fun clashing between legacy props and our custom props
  // @ts-ignore
  width: ['18px', '16px'],
  // @ts-ignore
  height: ['18px', '16px'],
};

export interface BaseCheckboxProps extends React.InputHTMLAttributes<HTMLInputElement> {
  indeterminate?: boolean;
  checked?: boolean;
  css?: any;
}
const BaseCheckbox = React.forwardRef<HTMLInputElement, BaseCheckboxProps>(
  ({ indeterminate, checked, ...restProps }, ref) => (
    <Wrapper>
      <CheckboxInput
        ref={ref}
        type="checkbox"
        indeterminate={indeterminate}
        checked={checked}
        {...restProps}
      />
      <CheckboxCheckmarkWrapper indeterminate={indeterminate}>
        <StyledIcon size="0.65rem" />
        {indeterminate && !checked && <CheckboxIndeterminate />}
      </CheckboxCheckmarkWrapper>
    </Wrapper>
  ),
);

export interface CheckboxProps extends BaseCheckboxProps {
  label?: React.ReactNode;
  description?: React.ReactNode;
  css?: any;
}
const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
  ({ label, id, description, disabled, ...restProps }, ref) => {
    return (
      <React.Fragment>
        <CheckboxLabel htmlFor={id} description={description} disabled={disabled}>
          <BaseCheckbox ref={ref} id={id} {...restProps} disabled={disabled} />
          {label}
        </CheckboxLabel>
      </React.Fragment>
    );
  },
);

export { Checkbox, CheckboxLabel, BaseCheckbox };
