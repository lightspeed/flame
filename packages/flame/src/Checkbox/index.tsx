import {
  BaseCheckbox,
  BaseCheckboxProps,
  Checkbox,
  CheckboxProps,
  CheckboxLabel,
} from './Checkbox';

export { BaseCheckbox, BaseCheckboxProps, Checkbox, CheckboxProps, CheckboxLabel };
