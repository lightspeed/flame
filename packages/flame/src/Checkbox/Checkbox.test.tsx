/* eslint-disable import/no-extraneous-dependencies */
import React, { FormEventHandler } from 'react';
import { createComponent, customRender, fireEvent } from 'test-utils';
import { Checkbox } from './Checkbox';

type RenderTest = {
  when: string;
  props: {
    checked?: boolean;
    onChange?: FormEventHandler<any>;
    disabled?: boolean;
    indeterminate?: boolean;
    id?: string;
    value?: string;
    label?: any;
    description?: any;
  };
};

describe('<Checkbox />', () => {
  describe('Snapshots', () => {
    const renderTests: RenderTest[] = [
      { when: 'empty props', props: {} },
      { when: 'not checked', props: { checked: false } },
      { when: 'checked', props: { checked: true, onChange: () => {} } },
      { when: 'disabled', props: { disabled: true } },
      { when: 'indeterminate', props: { indeterminate: true } },
      {
        when: 'all props',
        props: { id: 'id', value: 'value', label: 'Label', description: 'Description' },
      },
      {
        when: 'all props and label/description with html',
        props: {
          id: 'id',
          value: 'value',
          label: <div>Label</div>,
          description: <div>Description</div>,
        },
      },
    ];
    renderTests.forEach(test => {
      describe(`when ${test.when}`, () => {
        it('should render correctly', () => {
          const component = createComponent(<Checkbox {...test.props} />);
          expect(component.toJSON()).toMatchSnapshot();
        });
      });
    });
  });

  describe('Events', () => {
    it('handles the `onChange` event', () => {
      const onChange = jest.fn();
      const { container } = customRender(<Checkbox onChange={onChange} />);
      fireEvent.click(container.querySelector('input[type="checkbox"]'));
      expect(onChange).toHaveBeenCalled();
    });

    it('handles the `onFocus` event', () => {
      const onFocus = jest.fn();
      const { container } = customRender(<Checkbox onFocus={onFocus} />);
      fireEvent.focus(container.querySelector('input[type="checkbox"]'));
      expect(onFocus).toHaveBeenCalled();
    });

    it('handles the `onBlur` event', () => {
      const onBlur = jest.fn();
      const { container } = customRender(<Checkbox onBlur={onBlur} />);
      fireEvent.blur(container.querySelector('input[type="checkbox"]'));
      expect(onBlur).toHaveBeenCalled();
    });
  });

  describe('Render', () => {
    it('renders with a className attribute', () => {
      const className = 'custom-class';
      const { container } = customRender(<Checkbox className={className} />);

      expect(container.querySelectorAll('.custom-class')).toBeTruthy();
    });

    it('renders custom components in Label and Description', () => {
      const { getByText } = customRender(
        <Checkbox label={<div>my label</div>} description={<div>my description</div>} />,
      );

      getByText('my label');
      getByText('my description');
    });
  });
});
