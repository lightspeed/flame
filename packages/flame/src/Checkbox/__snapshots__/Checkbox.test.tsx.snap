// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Snapshots when all props and label/description with html should render correctly 1`] = `
.emotion-9 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-9 {
    font-size: 0.875rem;
  }
}

.emotion-7 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-6 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:checked + .emotion-6 .eo2dg7r2 {
  display: none;
}

.emotion-0:checked + div .emotion-2 {
  display: block;
}

.emotion-0:not(:checked) + div .emotion-2 {
  display: none;
}

.emotion-0:focus + .emotion-6 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-6 {
  opacity: 0.5;
}

.emotion-5 {
  position: relative;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 0.1875rem;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  cursor: default;
  background: linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-5 {
    width: 16px;
    height: 16px;
  }
}

.emotion-3 {
  vertical-align: text-bottom;
  width: 0.65rem;
  height: 0.65rem;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  fill: #fff;
}

.emotion-11 {
  margin: 0;
  margin-left: 1.85rem;
  font-size: 0.875rem;
  line-height: 1.125rem;
}

@media screen and (min-width:601px) {
  .emotion-11 {
    margin-left: 1.75rem;
  }
}

<div
  className="emotion-13 emotion-14"
>
  <label
    className="emotion-9 emotion-10"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
    htmlFor="id"
  >
    <div
      className="emotion-7 emotion-8"
    >
      <input
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        id="id"
        type="checkbox"
        value="value"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-5 emotion-6"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <svg
          className="emotion-2 cr-icon emotion-3 emotion-4"
          height="0.65rem"
          viewBox="0 0 16 16"
          width="0.65rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="cr-icon__base cr-icon__base-1"
            d="M5.662 12.508c.656.686 1.735.65 2.345-.079l5.6-6.68c.58-.692.51-1.74-.157-2.342a1.561 1.561 0 0 0-2.259.164L6.802 8.81 4.738 6.822a1.56 1.56 0 0 0-2.264-.01 1.704 1.704 0 0 0-.01 2.347l3.198 3.348z"
          />
        </svg>
      </div>
    </div>
    <div>
      Label
    </div>
  </label>
  <div
    className="emotion-11 emotion-12"
    size="small"
  >
    <div>
      Description
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots when all props should render correctly 1`] = `
.emotion-9 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-9 {
    font-size: 0.875rem;
  }
}

.emotion-7 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-6 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:checked + .emotion-6 .eo2dg7r2 {
  display: none;
}

.emotion-0:checked + div .emotion-2 {
  display: block;
}

.emotion-0:not(:checked) + div .emotion-2 {
  display: none;
}

.emotion-0:focus + .emotion-6 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-6 {
  opacity: 0.5;
}

.emotion-5 {
  position: relative;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 0.1875rem;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  cursor: default;
  background: linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-5 {
    width: 16px;
    height: 16px;
  }
}

.emotion-3 {
  vertical-align: text-bottom;
  width: 0.65rem;
  height: 0.65rem;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  fill: #fff;
}

.emotion-11 {
  margin: 0;
  margin-left: 1.85rem;
  font-size: 0.875rem;
  line-height: 1.125rem;
}

@media screen and (min-width:601px) {
  .emotion-11 {
    margin-left: 1.75rem;
  }
}

<div
  className="emotion-13 emotion-14"
>
  <label
    className="emotion-9 emotion-10"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
    htmlFor="id"
  >
    <div
      className="emotion-7 emotion-8"
    >
      <input
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        id="id"
        type="checkbox"
        value="value"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-5 emotion-6"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <svg
          className="emotion-2 cr-icon emotion-3 emotion-4"
          height="0.65rem"
          viewBox="0 0 16 16"
          width="0.65rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="cr-icon__base cr-icon__base-1"
            d="M5.662 12.508c.656.686 1.735.65 2.345-.079l5.6-6.68c.58-.692.51-1.74-.157-2.342a1.561 1.561 0 0 0-2.259.164L6.802 8.81 4.738 6.822a1.56 1.56 0 0 0-2.264-.01 1.704 1.704 0 0 0-.01 2.347l3.198 3.348z"
          />
        </svg>
      </div>
    </div>
    Label
  </label>
  <div
    className="emotion-11 emotion-12"
    size="small"
  >
    Description
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots when checked should render correctly 1`] = `
.emotion-9 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-9 {
    font-size: 0.875rem;
  }
}

.emotion-7 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-6 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:checked + .emotion-6 .eo2dg7r2 {
  display: none;
}

.emotion-0:checked + div .emotion-2 {
  display: block;
}

.emotion-0:not(:checked) + div .emotion-2 {
  display: none;
}

.emotion-0:focus + .emotion-6 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-6 {
  opacity: 0.5;
}

.emotion-5 {
  position: relative;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 0.1875rem;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  cursor: default;
  background: linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-5 {
    width: 16px;
    height: 16px;
  }
}

.emotion-3 {
  vertical-align: text-bottom;
  width: 0.65rem;
  height: 0.65rem;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  fill: #fff;
}

<div
  className="emotion-11 emotion-12"
>
  <label
    className="emotion-9 emotion-10"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
  >
    <div
      className="emotion-7 emotion-8"
    >
      <input
        checked={true}
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        onChange={[Function]}
        type="checkbox"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-5 emotion-6"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <svg
          className="emotion-2 cr-icon emotion-3 emotion-4"
          height="0.65rem"
          viewBox="0 0 16 16"
          width="0.65rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="cr-icon__base cr-icon__base-1"
            d="M5.662 12.508c.656.686 1.735.65 2.345-.079l5.6-6.68c.58-.692.51-1.74-.157-2.342a1.561 1.561 0 0 0-2.259.164L6.802 8.81 4.738 6.822a1.56 1.56 0 0 0-2.264-.01 1.704 1.704 0 0 0-.01 2.347l3.198 3.348z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`<Checkbox /> Snapshots when disabled should render correctly 1`] = `
.emotion-9 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-9 {
    font-size: 0.875rem;
  }
}

.emotion-7 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-6 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:checked + .emotion-6 .eo2dg7r2 {
  display: none;
}

.emotion-0:checked + div .emotion-2 {
  display: block;
}

.emotion-0:not(:checked) + div .emotion-2 {
  display: none;
}

.emotion-0:focus + .emotion-6 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-6 {
  opacity: 0.5;
}

.emotion-5 {
  position: relative;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 0.1875rem;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  cursor: default;
  background: linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-5 {
    width: 16px;
    height: 16px;
  }
}

.emotion-3 {
  vertical-align: text-bottom;
  width: 0.65rem;
  height: 0.65rem;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  fill: #fff;
}

<div
  className="emotion-11 emotion-12"
>
  <label
    className="emotion-9 emotion-10"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
  >
    <div
      className="emotion-7 emotion-8"
    >
      <input
        className="emotion-0 emotion-1"
        disabled={true}
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        type="checkbox"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-5 emotion-6"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <svg
          className="emotion-2 cr-icon emotion-3 emotion-4"
          height="0.65rem"
          viewBox="0 0 16 16"
          width="0.65rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="cr-icon__base cr-icon__base-1"
            d="M5.662 12.508c.656.686 1.735.65 2.345-.079l5.6-6.68c.58-.692.51-1.74-.157-2.342a1.561 1.561 0 0 0-2.259.164L6.802 8.81 4.738 6.822a1.56 1.56 0 0 0-2.264-.01 1.704 1.704 0 0 0-.01 2.347l3.198 3.348z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`<Checkbox /> Snapshots when empty props should render correctly 1`] = `
.emotion-9 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-9 {
    font-size: 0.875rem;
  }
}

.emotion-7 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-6 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:checked + .emotion-6 .eo2dg7r2 {
  display: none;
}

.emotion-0:checked + div .emotion-2 {
  display: block;
}

.emotion-0:not(:checked) + div .emotion-2 {
  display: none;
}

.emotion-0:focus + .emotion-6 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-6 {
  opacity: 0.5;
}

.emotion-5 {
  position: relative;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 0.1875rem;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  cursor: default;
  background: linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-5 {
    width: 16px;
    height: 16px;
  }
}

.emotion-3 {
  vertical-align: text-bottom;
  width: 0.65rem;
  height: 0.65rem;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  fill: #fff;
}

<div
  className="emotion-11 emotion-12"
>
  <label
    className="emotion-9 emotion-10"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
  >
    <div
      className="emotion-7 emotion-8"
    >
      <input
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        type="checkbox"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-5 emotion-6"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <svg
          className="emotion-2 cr-icon emotion-3 emotion-4"
          height="0.65rem"
          viewBox="0 0 16 16"
          width="0.65rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="cr-icon__base cr-icon__base-1"
            d="M5.662 12.508c.656.686 1.735.65 2.345-.079l5.6-6.68c.58-.692.51-1.74-.157-2.342a1.561 1.561 0 0 0-2.259.164L6.802 8.81 4.738 6.822a1.56 1.56 0 0 0-2.264-.01 1.704 1.704 0 0 0-.01 2.347l3.198 3.348z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`<Checkbox /> Snapshots when indeterminate should render correctly 1`] = `
.emotion-11 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-11 {
    font-size: 0.875rem;
  }
}

.emotion-9 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-8 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:checked + .emotion-8 .emotion-6 {
  display: none;
}

.emotion-0:checked + div .emotion-2 {
  display: block;
}

.emotion-0:not(:checked) + div .emotion-2 {
  display: none;
}

.emotion-0:focus + .emotion-8 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-8 {
  opacity: 0.5;
}

.emotion-3 {
  vertical-align: text-bottom;
  width: 0.65rem;
  height: 0.65rem;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  fill: #fff;
}

.emotion-7 {
  position: relative;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 0.1875rem;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  cursor: default;
  background: linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
  border-color: #2875c6;
}

@media screen and (min-width:601px) {
  .emotion-7 {
    width: 16px;
    height: 16px;
  }
}

.emotion-5 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 0.5rem;
  height: 0.125rem;
  border-radius: 0.1875rem;
  background: #1d5794;
}

<div
  className="emotion-13 emotion-14"
>
  <label
    className="emotion-11 emotion-12"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
  >
    <div
      className="emotion-9 emotion-10"
    >
      <input
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        type="checkbox"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-7 emotion-8"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <svg
          className="emotion-2 cr-icon emotion-3 emotion-4"
          height="0.65rem"
          viewBox="0 0 16 16"
          width="0.65rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="cr-icon__base cr-icon__base-1"
            d="M5.662 12.508c.656.686 1.735.65 2.345-.079l5.6-6.68c.58-.692.51-1.74-.157-2.342a1.561 1.561 0 0 0-2.259.164L6.802 8.81 4.738 6.822a1.56 1.56 0 0 0-2.264-.01 1.704 1.704 0 0 0-.01 2.347l3.198 3.348z"
          />
        </svg>
        <div
          className="emotion-5 emotion-6"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`<Checkbox /> Snapshots when not checked should render correctly 1`] = `
.emotion-9 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-9 {
    font-size: 0.875rem;
  }
}

.emotion-7 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-6 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:checked + .emotion-6 .eo2dg7r2 {
  display: none;
}

.emotion-0:checked + div .emotion-2 {
  display: block;
}

.emotion-0:not(:checked) + div .emotion-2 {
  display: none;
}

.emotion-0:focus + .emotion-6 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-6 {
  opacity: 0.5;
}

.emotion-5 {
  position: relative;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 0.1875rem;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  cursor: default;
  background: linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-5 {
    width: 16px;
    height: 16px;
  }
}

.emotion-3 {
  vertical-align: text-bottom;
  width: 0.65rem;
  height: 0.65rem;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  fill: #fff;
}

<div
  className="emotion-11 emotion-12"
>
  <label
    className="emotion-9 emotion-10"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
  >
    <div
      className="emotion-7 emotion-8"
    >
      <input
        checked={false}
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        type="checkbox"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-5 emotion-6"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <svg
          className="emotion-2 cr-icon emotion-3 emotion-4"
          height="0.65rem"
          viewBox="0 0 16 16"
          width="0.65rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="cr-icon__base cr-icon__base-1"
            d="M5.662 12.508c.656.686 1.735.65 2.345-.079l5.6-6.68c.58-.692.51-1.74-.157-2.342a1.561 1.561 0 0 0-2.259.164L6.802 8.81 4.738 6.822a1.56 1.56 0 0 0-2.264-.01 1.704 1.704 0 0 0-.01 2.347l3.198 3.348z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;
